scan/src/pages/DataSourceIndexPage.re

```diff
@@ -16,91 +16,101 @@ module Styles = {
 [@react.component]
 let make = (~dataSourceID, ~hashtag: Route.data_source_tab_t) => {
   let dataSourceSub = DataSourceSub.get(dataSourceID);
-  <Section pbSm=0>
-    <div className=CssHelper.container>
-      <Row.Grid marginBottom=40 marginBottomSm=16>
-        <Col.Grid>
-          <Heading value="Data Source" size=Heading.H4 marginBottom=40 marginBottomSm=24 />
-          {switch (dataSourceSub) {
-           | Data({id, name}) =>
-             <div className={CssHelper.flexBox()}>
-               <TypeID.DataSource id position=TypeID.Title />
-               <HSpacing size=Spacing.sm />
-               <Heading size=Heading.H3 value=name />
-             </div>
-           | _ => <LoadingCensorBar width=270 height=15 />
-           }}
-        </Col.Grid>
-      </Row.Grid>
-      <Row.Grid marginBottom=24>
-        <Col.Grid>
-          <div className=Styles.infoContainer>
-            <Heading value="Information" size=Heading.H4 style=Styles.infoHeader marginBottom=24 />
-            <div className={CssHelper.flexBox()}>
-              <Heading value="Owner" size=Heading.H5 />
-              <HSpacing size=Spacing.xs />
-              <CTooltip tooltipText="The owner of the data source">
-                <Icon name="fal fa-info-circle" size=10 />
-              </CTooltip>
-            </div>
-            <VSpacing size=Spacing.sm />
-            {switch (dataSourceSub) {
-             | Data({owner}) => <AddressRender address=owner position=AddressRender.Subtitle />
-             | _ => <LoadingCensorBar width=284 height=15 />
-             }}
-            <VSpacing size=Spacing.lg />
-            <Heading value="Description" size=Heading.H5 marginBottom=16 />
+
+  switch (dataSourceSub) {
+  | NoData => <NotFound />
+  | _ =>
+    <Section pbSm=0>
+      <div className=CssHelper.container>
+        <Row.Grid marginBottom=40 marginBottomSm=16>
+          <Col.Grid>
+            <Heading value="Data Source" size=Heading.H4 marginBottom=40 marginBottomSm=24 />
```

Comment:
Can we move all of the elements from the 23rd line as a component to make it easier to maintain?
@evilpeach What do you think?

Reply:
Can be made in the same file

Reply:
Updated! Please check again. @pzshine